### PR TITLE
corrected errors

### PR DESCRIPTION
**Pull request by timur-rimsky**
I've corrected errors
made some changes in gem file
so specs pass

### CODE REVIEW

**Comment by Valve** [comment]
@timur-rimsky mention me like that @Valve so that I get notifications.

File: spec/mxmnd_spec.rb [hunk 1]
```diff
@@ -1,7 +1,64 @@
...
 describe Mxmnd do
   it 'raises BadRequest on missing IP address' do
-    expect { Mxmnd::city(nil) }.to raise_error Mxmnd::BadRequest
+    expect { Mxmnd.city(nil) }.to raise_error Mxmnd::BadRequest
+  end
+end
+
+describe 'Mxmnd.city response' do
+  before {
+  	url = 'https://geoip.maxmind.com/geoip/v2.1/city/45.17.22.11'
```
Comment (Valve) [inline spec/mxmnd_spec.rb]: indentation is off here

File: spec/mxmnd_spec.rb [hunk 2]
```diff
@@ -1,7 +1,64 @@
...
+    path = './spec/features/mxmnd_res.txt'
```
Comment (Valve) [inline spec/mxmnd_spec.rb]: better be more specific and call it `response_path`

File: spec/mxmnd_spec.rb [hunk 3]
```diff
@@ -1,7 +1,64 @@
...
+    ruby_headers = {
```
Comment (Valve) [inline spec/mxmnd_spec.rb]: these headers are not ruby-specific, so better just call them `headers` or `request headers`

File: spec/mxmnd_spec.rb [hunk 4]
```diff
@@ -1,7 +1,64 @@
...
+      'Accept-Encoding' => 'gzip;q=1.0,deflate;q=0.6,identity;q=0.3',
+       'User-Agent' => 'Ruby'
+    }
+    # i've changed "Content-Length" property of recorded response from 1476 to 2515
+    # because of by unknown reason stub didn't return the whole response
+    response_file = File.new(path)
+    stub_request(:get, 'http://https//geoip.maxmind.com/geoip/v2.1/city/45.17.22.11:80/').
+      with(headers: ruby_headers).
+      to_return(response_file)
+    @response = JSON.parse(Net::HTTP.get(url, '/'))
```
Comment (Valve) [inline spec/mxmnd_spec.rb]: the use of instance variables in specs is discouraged, Please use `let` for this.
e.g.
```
let(:response) { JSON.parse... }
```

Then you can just use `response` variable inside your spec examples.

File: spec/mxmnd_spec.rb [hunk 5]
```diff
@@ -1,7 +1,64 @@
...
+  }
+
+  it 'should be a hash' do
+    expect(@response.is_a?(Hash)).to be(true)
+  end
+
+  it 'should have distinct keys' do
+    keys_array = [
```
Comment (Valve) [inline spec/mxmnd_spec.rb]: keys (anything plural is array be definition)

File: spec/mxmnd_spec.rb [hunk 6]
```diff
@@ -1,7 +1,64 @@
...
+      expect(hash_including_locales(@response['continent'], 8)).to be(true)
+    end
+    it 'key country should be a hash including 8 locales' do
+      expect(hash_including_locales(@response['country'], 8)).to be(true)
+    end
+    it 'key registered_country should be a hash including 8 locales' do
+      expect(hash_including_locales(@response['registered_country'], 8)).to be(true)
+    end
+    it 'key subdivisions should be an array' do
+      expect(@response['subdivisions'].is_a?(Array)).to be(true)
```
Comment (Valve) [inline spec/mxmnd_spec.rb]: there should be a matcher that does that in one check, eg:

```
expect(obj).to be_a(type)
```

File: spec/spec_helper.rb [hunk 7]
```diff
@@ -1,2 +1,12 @@
...
+$LOAD_PATH.unshift File.expand_path('../../lib', __FILE__)
+require 'mxmnd'
+require 'webmock/rspec'
+
+def presence_of_keys?(dict, keys_array)
+  keys_array.each { |key| return false unless dict[key] }
+  true
+end
+
+def hash_including_locales(dict, num_of_locales)
```
Comment (Valve) [inline spec/spec_helper.rb]: this method doesn't care if second argument is locales or something else, a better way is to just bring this into the spec example, e.g. `expect(dict['name'].size).to eq 8`.
Any custom matching methods make spec reading more confusing.

**Review by Valve — CHANGES_REQUESTED** [review]
Overall it's much better this time.
I left some comments regarding the stylistics of your code.

You can also be more specific about your spec expectation, instead of saying that name should have 8 locales, you can just directly verify that a name is %w[a b c].
But overall it's a good progress.

**Comment by timur-rimsky** [comment]
@Valve, I've made corrections according to your notices. 
Please look it for mistakes.

**Pull request merged**